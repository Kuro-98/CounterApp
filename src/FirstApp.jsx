import PropTypes from 'prop-types';

export const App = ({ title, subTitle, name }) => {
  /* Existe otra manera de definir el valor por defecto
  de los props y no hacerlo directo aqui arriba asi que 
  veremos la mejor opcion cuando tenemos muchos: */
  return (
    <>
      <div data-testid='test-title'>{title}</div>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

App.propTypes = {
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
};

App.defaultProps = {
  name: 'Manolo Morales',
  subTitle: 'No hay subtitulo',
  //title: 'No hay titulo',
};

/* Usualmente estas dos configuraciones deben ponerse al final*/
