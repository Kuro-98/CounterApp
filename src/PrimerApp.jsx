//Debemos intentar no meter variables ni funciones
/*Dentro del componente a menos que sea sumamente necesario
como en el caso de useState o cosas que forcen usar algo
dentro del componente.*/

export const PrimerApp = () => {
  return (
    <>
      <h2>"Hola como estas tu"</h2>
      <p>Soy un subtitulo</p>
    </>
  );
};

// {JSON.stringify(nombreDarreglo)}
/*Nos permite forzar objetos a texto para imprimirlos ,
pero generalmente no debemos mandar a llamar
objetos dentro del jsx*/
