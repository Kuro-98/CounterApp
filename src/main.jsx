import React from 'react';

//comenzamos poniendo imr para el primer import que se ve arriba

import ReactDOM from 'react-dom/client';
//siempre siempre deben usarse estos dos imports.
import { App } from './FirstApp';
import { PrimerApp } from './PrimerApp';
import './styles.css';
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={20} />
    {/* <App title='Hola, soy Manolo' /> */}
  </React.StrictMode>
);

//siempre debe existir una forma de renderizar los comps

//Esta seria la estructura necesaria para renderizar
//pero esto no se debe hacer.
