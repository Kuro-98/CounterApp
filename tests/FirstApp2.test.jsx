import { render, screen } from '@testing-library/react';
import { App } from '../src/FirstApp';

describe('Pruebas en FirstApp', () => {
  const title = 'Hola Soy Kuro';

  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<App title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el mensaje "Hola, Soy Kuro" ', () => {
    render(<App title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Debe de mostrar el titulo en un h1', () => {
    render(<App title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('Debe de mostrar el subtitulo enviado por props', () => {
    render(<App title='Sop bro' subTitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
