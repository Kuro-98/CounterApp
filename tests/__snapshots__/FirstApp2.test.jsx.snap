// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola Soy Kuro
  </div>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Manolo Morales
  </p>
</div>
`;
